release-notes.txt

```diff
@@ -1,3 +1,705 @@
+        * assert CPU is defined by BOARD/Makefile.features
+        + add TERMFLASHDEPS to TERMDEPS so it also applies to `test`
+        * tools: allow make reset via avrdude
+        * tools: set DEBUG_ADAPTER_ID as JLINK_SERIAL
+
+Testing
+-------
+    - dist/tests/philip: Removed old code and tests
+    * dist/tools/testrunner:
+        * make interactive test sync retries/delay configurable
+        * reset before opening terminal
+        - reset after opening terminal only if interactive test sync is not used
+        * add variable for customizing a delay before reset
+        * make interactive test sync retries/delay configurable
+    * dist/tools/compile_and_test_for_board.py: allow use of wildcards for applications selection
+    + murdock: enable esp32-wroom-32 for CI testing
+    * tests:
+        + use test_interactive_test_util
+        + add automated tests guidelines in README
+        - remove APPLICATION definition in tests/*
+    + tests/gnrc_sock_neterr: proof of concept test for gnrc_sock + gnrc_neterr
+    + tests/bench_xtimer: initial import
+    * tests/lwip: enable IPv4 
+    * tests/malloc: improve application and add an automatic test script
+    * tests/memarray: add python script for automatic testing
+    * tests/nordic_softdevice: cleanup test script and fix test
+    + tests/periph_timer_short_relative_set: initial import
+    * tests/periph_cpuid: add automatic test script
+    * tests/sys_arduino: fix test synchronization issues
+    * tests/unittests: fix `tests-pkt` for non-32bit platforms
+
+
+API Changes
+========================
+
+- drivers/ina220: major changes to comply with RIOT's driver design goals & SAUL adaption 
+- usbus: unified terminology within USBUS to match the terminology from the specs.
+         `hdr`, `hdrs`, `headers`, etc is replaced everywhere with `descr` `descriptors` and `descriptors`.
+- Makefile.include: fail by default when errors are expected
+
+```diff
+/* change return type of `adc_sample()` to `int32_t` to allow support for ADCs
+   with resolution of more than 16 bits on 8 bit and 16 bit platforms*/
+- int adc_sample(adc_t line, adc_res_t res)
+= int32_t adc_sample(adc_t line, adc_res_t res)
+
+/* pass memo to resp_handler directly  */
+- typedef void (*gcoap_resp_handler_t)(unsigned req_state, coap_pkt_t* pdu,
+-                                      sock_udp_ep_t *remote);
++ typedef void (*gcoap_resp_handler_t)(const gcoap_request_memo_t *memo,
++                                      coap_pkt_t* pdu,
++                                      const sock_udp_ep_t *remote);
+
+/* allow to pass user context to requests */
+  size_t gcoap_req_send(const uint8_t *buf, size_t len,
+                        const sock_udp_ep_t *remote,
+-                       gcoap_resp_handler_t resp_handler);
++                       gcoap_resp_handler_t resp_handler, void *context);
+
+  static inline size_t gcoap_req_send2(const uint8_t *buf, size_t len,
```

Comment:
Please remove. gcoap_req_send2() was removed in #12837 for this release,

Reply:
True, sorry about that. removed.

Reply:
No worries. IIRC you should be approaching the "please let this end" phase of release management. :-)